_tasks/sync-proposal-data.mjs

```diff
@@ -0,0 +1,378 @@
+// Helpers
+// -----------------------------------------------------------------------------
+
+/**
+ * Gets the URL value from the Markdown link key in the proposals README.
+ * @param {string} linkLabelLike
+ * @returns {string}
+ */
+function getUrlFromDoc(linkLabelLike) {
+  // log.debug('getUrlFromDoc linkLabelLike: ' + linkLabelLike);
+  const linkLabelLikeRegex = new RegExp(`\\[${linkLabelLike}.*\\]: (.*)`, 'g');
+  // log.debug({ linkLabelLikeRegex });
+  const results = linkLabelLikeRegex.exec(prpslsReadmeMarkdown);
+  // log.debug({ results });
+  return results[1];
+}
+
+/**
+ * Downloads the proposal's README file using the GitHub API, parses the
+ * Markdown file, and returns the contents of the JavaScript code block.
+ *
+ * If there are multiple matching JavaScript code blocks, the one that comes
+ * last in the document is used. If there are no matching JavaScript code
+ * blocks, an empty string will be used. (In such cases, a warning is emitted.)
+ *
+ * @param {string} prpslId
+ * @returns {string}
+ */
+async function getCodeSample(prpslId) {
+  const codeBlocks = [];
+  try {
+    // Some proposal repos use a different branch (e.g., `gh-pages`) for their
+    // README file. Because of this, we need repo metadata to find the raw URL.
+    let rawReadmeUrl;
+    await ghFileImporter
+      .fetchPathMetadata('tc39', prpslId, 'README.md')
+      .then(function (res) {
+        // log.debug(res.data.download_url);
+        rawReadmeUrl = res.data.download_url;
+      });
+
+    // Get the file and parse out the code samples.
+    await ghFileImporter
+      .importFileFromUrl(rawReadmeUrl)
+      .then(async function (res) {
+        const docSourceText = res;
+        await unified()
+          .use(markdown)
+          .use(function () {
+            return function (tree) {
+              visit(tree, null, (node, index) => {
+                if (
+                  node.type === 'code' &&
+                  node.lang.match(REGEX_JS_INFO_STRS).length > 0
+                ) {
+                  codeBlocks.push(he.encode(node.value));
+                }
+              });
+            };
+          })
+          .use(remark2rehype, { allowDangerousHtml: true })
+          .use(raw)
+          .use(htmlStringify)
+          .process(docSourceText);
+      });
+  } catch (err) {
+    log.error(err);
+    throw err;
+  }
+
+  return codeBlocks.length > 0 ? codeBlocks.pop() : '';
+}
+
+/**
+ * A 'collapsed reference link' (`[foo][]`) consists of a link label that
+ * matches a 'link reference definition' elsewhere in the document, followed by
+ * the string `[]`.
+ * @param {string} linkLabelLike
+ * @returns {string} The proposal ID.
+ */
+function getPrpslId(linkLabelLike) {
+  let prpslUrl = getUrlFromDoc(linkLabelLike);
+  if (prpslUrl.endsWith('/')) {
+    prpslUrl = prpslUrl.slice(0, -1); // Ditch trailing slashes & prevent crash.
+  }
+  return prpslUrl.split('/').pop().toLowerCase();
+}
+
+// urlKeyLike -> linkLabelLike
```

Comment:
```suggestion
// urlKeyLike -> linkTextLike
```

I left it like this for so long because this isn't the actual solution. More soon.

Reply:
Okay, here's the explanation for this.

We wouldn't really need this variable if we had something like a [`w3c.json`](https://github.com/w3c/webref/blob/master/w3c.json) file in each of the proposal repositories. It would be a proposal metadata file that would contain much of what's missing here.

It's not strictly necessary, but would be nice to be able to know where to reliably find (at least) the proposal description which currently is quite complex to determine (I could write a NLP algorithm to locate the most appropriate sentence in each repo's `README.md` file).